### DISCUSSION TITLE
Specify entity graph for save operation

### DISCUSSION THREAD

**Discussion by mr-nothing** [Cosium/spring-data-jpa-entity-graph#157] [Q&A] [answered]
I wonder is it by design that library doesn't provide `save` methods that contain EntityGraph as a second parameter? Of course I can do additional find operation but I think it could be nice to be able to control what relations I fetch with the entity I just saved.

In the code it could look like this:
```
    ...
    repository.save(entity, DynamicEntityGraph.fetching("child1"))
    ...
```

Thanks in advance for your help!

**Comment by reda-alaoui** [comment, marked as answer]
Behind the scene, AFAIK, spring data jpa `save` operation only invokes `EntityManager#persist` or `EntityManager#merge`. AFAIK, none of the latters accept `EntityGraph` attributes.

**Reply by mr-nothing** [reply]
Ah, silly me, you are right. Thanks for clarification!
